src/js/letters/components/Address.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import _ from 'lodash';
+import { set } from 'lodash/fp';
+
+import ErrorableSelect from './ErrorableSelect';
+import ErrorableTextInput from './ErrorableTextInput';
+import { isNotBlank, isBlankAddress, isValidUSZipCode, isValidCanPostalCode } from '../../common/utils/validations';
+import { countries, states } from '../../common/utils/options-for-select';
+
+/**
+ * Input component for an address.
+ *
+ * No validation is provided through a currently stubbed isAddressValid function.
+ */
+class Address extends React.Component {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.validateAddressField = this.validateAddressField.bind(this);
+    this.validatePostalCode = this.validatePostalCode.bind(this);
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('address-input-');
+  }
+
+  // TODO: Look into if this is the best way to update address,
+  // it is incredibly slow right now
+  handleChange(path, update) {
+    let address = set(path, update, this.props.value);
+    // if country is changing we should clear the state
+    if (path === 'country') {
+      address = set('state', '', address);
+    }
+
+    this.props.onUserInput(address);
+  }
+
+  validateAddressField(field) {
+    if (this.props.required || !isBlankAddress(this.props.value)) {
+      return isNotBlank(field);
+    }
+
+    return true;
+  }
+
+  validatePostalCode(postalCodeField) {
```

Comment:
Since this returns a `boolean`, I'd recommend a rename to `isValidPostalCode`.  Same goes for `validateAddressField`

Reply:
Good point. I can make that change.